beacon-chain/sync/rpc_beacon_blocks_by_root.go

```diff
@@ -114,41 +116,78 @@ func (s *Service) beaconBlocksRootRPCHandler(ctx context.Context, msg interface{
 	return nil
 }
 
-func (s *Service) requestPendingBlobs(ctx context.Context, b interfaces.ReadOnlyBeaconBlock, br []byte, id peer.ID) error {
-	// Block before deneb has no blob.
-	if b.Version() < version.Deneb {
-		return nil
+// requestPendingBlobs handles the request for pending blobs based on the given beacon block.
+func (s *Service) requestPendingBlobs(ctx context.Context, block interfaces.ReadOnlyBeaconBlock, blockRoot []byte, peerID peer.ID) error {
+	if block.Version() < version.Deneb {
+		return nil // Block before deneb has no blob.
 	}
-	c, err := b.Body().BlobKzgCommitments()
+
+	commitments, err := block.Body().BlobKzgCommitments()
 	if err != nil {
 		return err
 	}
-	// No op if the block has no blob commitments.
-	if len(c) == 0 {
-		return nil
+
+	if len(commitments) == 0 {
+		return nil // No operation if the block has no blob commitments.
 	}
 
-	// Build request for blob sidecars.
-	blobId := make([]*eth.BlobIdentifier, len(c))
-	for i := range c {
-		blobId[i] = &eth.BlobIdentifier{Index: uint64(i), BlockRoot: br}
+	contextByte, err := ContextByteVersionsForValRoot(s.cfg.chain.GenesisValidatorsRoot())
+	if err != nil {
+		return err
 	}
 
-	ctxByte, err := ContextByteVersionsForValRoot(s.cfg.chain.GenesisValidatorsRoot())
+	request, err := s.constructPendingBlobsRequest(ctx, bytesutil.ToBytes32(blockRoot), len(commitments))
```

Comment:
I would change this function to take the `[32]byte` that `HashTreeRoot` returned anyway, instead of passing the slice.